bob-backend/src/core.rs

```diff
@@ -335,15 +337,23 @@ impl Backend {
     }
 
     pub async fn get(&self, key: BobKey, options: &BobOptions) -> Result<BobData, Error> {
-        let (vdisk_id, disk_path) = self.mapper.get_operation(key);
+        let (vdisk_id, disk_paths) = self.mapper.get_operation(key);
 
         // we cannot get data from alien if it belong this node
 
         if options.get_normal() {
-            if let Some(path) = disk_path {
+            if let Some(paths) = disk_paths {
                 trace!("GET[{}] try read normal", key);
-                self.get_local(key, Operation::new_local(vdisk_id, path.clone()))
-                    .await
+                let mut futures: FuturesUnordered<_> = paths.into_iter().map(|path| {
+                    self.get_local(key, Operation::new_local(vdisk_id, path))
+                }).collect();
+                while let Some(result) = futures.next().await {
+                    match result {
+                        Ok(data) => return Ok(data),
+                        Err(_) => continue,
```

Comment:
You lost particular error here and silently convert it to `internal` at the end. This is not correct